sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -241,23 +241,24 @@ def from_blob_url(cls, blob_url, credential=None, snapshot=None, **kwargs):
         account_path = ""
         if ".core." in parsed_url.netloc:
             # .core. is indicating non-customized url. Blob name with directory info can also be parsed.
-            path_blob = parsed_url.path.lstrip('/').split('/', 1)
+            path_blob = parsed_url.path.lstrip('/').split('/', maxsplit=1)
         elif "localhost" in parsed_url.netloc or "127.0.0.1" in parsed_url.netloc:
-            path_blob = parsed_url.path.lstrip('/').split('/', 2)
+            path_blob = parsed_url.path.lstrip('/').split('/', maxsplit=2)
             account_path += '/' + path_blob[0]
         else:
             # for customized url. blob name that has directory info cannot be parsed.
             path_blob = parsed_url.path.lstrip('/').split('/')
             if len(path_blob) > 2:
                 account_path = "/" + "/".join(path_blob[:-2])
+
         account_url = "{}://{}{}?{}".format(
             parsed_url.scheme,
             parsed_url.netloc.rstrip('/'),
             account_path,
             parsed_url.query)
-        container_name, blob_name = unquote(path_blob[-2]), unquote(path_blob[-1])
-        if not container_name or not blob_name:
```

Comment:
Can we leave this check in after line 261 in case the user passes an empty string for blob or container?

Reply:
Ok sure, I've put it back in with the duplicated error message refactored into a string variable.

I have not added a test for this scenario (empty blob/container string).
I tried a few different ways to feed in a bad blob_url, but couldn't find one that would:
- make it past the new check I put in above `unquote(path_blob[-2|-1]` call;
- then fail at `if not container_name|blob_name`.

Reply:
Thanks! Yeah, I'm not entirely sure if/how we could hit that case, but I think it's best to leave it in to be safe.